paimon-common/src/main/java/org/apache/paimon/fileindex/FileIndexOptions.java

```diff
@@ -44,13 +50,40 @@ public FileIndexOptions(long fileIndexInManifestThreshold) {
     }
 
     public void computeIfAbsent(String column, String indexType) {
```

Comment:
You should merge `CoreOptions.indexColumnsOptions` into this class.

And remove `isExternallyPerceptible`.